airflow/hooks/http_hook.py

```diff
@@ -0,0 +1,84 @@
+import logging
+
+import requests
+
+from airflow import utils
+from airflow.hooks.base_hook import BaseHook
+
+
+class HttpHook(BaseHook):
+
+    """
+    Interact with HTTP servers.
+    """
+
+    def __init__(
+            self, method='POST', http_conn_id='http_default'):
+        self.http_conn_id = http_conn_id
+        self.method = method
+
+    def get_conn(self, headers):
+        """
+        Returns http session for use with requests
+        """
+        conn = self.get_connection(self.http_conn_id)
+        session = requests.Session()
+        self.base_url = conn.host
+
+        if conn.port != None and conn.port > 0:
+            self.base_url = self.base_url + ":%d/"%( conn.port )
+
+        if len(conn.login) > 0:
+            session.auth = (conn.login, conn.password)
+        if headers != None:
+            session.headers.update(headers)
+
+        return session
+
+    def run(self, url, data=None, headers=None, extra_options={}):
+        """
+        Performs the request
```

Comment:
`endpoint` may be more clear than `url` as you're expecting only that part of the url here